src/sentry/api/endpoints/user_identity_details.py

```diff
@@ -14,5 +14,6 @@ class UserIdentityDetailsEndpoint(UserEndpoint):
     }
 
     def delete(self, request: Request, user, identity_id) -> Response:
-        AuthIdentity.objects.filter(user=user, id=identity_id).delete()
+        for ai in AuthIdentity.objects.filter(user=user, id=identity_id):
+            ai.delete()
```

Comment:
What's the impact of processing these rows sequentially vs using the queryset? I understand that we can't do this now due to the protections we have in place but I'm curious if there exist any users with a large number of auth identities.

Reply:
In practice, this will only have between 0 and 1 values -- it is equivalent to to a `get` with a `DoesNotExist` due to the `id` constraint.  The `user` constraint is a resource access constraint.  Using a loop is simply easier to write, but if it's hard to read I can adjust.

Reply:
Changed this to a get!